app/cdap/components/GridContainer/index.tsx

```diff
@@ -14,64 +14,119 @@
  * the License.
  */
 
-import React, { useState } from 'react';
+import React, { useState, useEffect } from 'react';
 import { connect, Provider } from 'react-redux';
 import DataPrepStore from 'components/DataPrep/store';
 import { execute } from 'components/DataPrep/store/DataPrepActionCreator';
-import DataPrepActions from 'components/DataPrep/store/DataPrepActions';
 import GridTable from 'components/GridTable';
-import { DATATYPE_OPTIONS } from 'components/WranglerGrid/NestedMenu/menuOptions/datatypeOptions';
+import { IMenuItem } from 'components/WranglerGrid/NestedMenu/MenuItemComponent';
+import Snackbar from 'components/Snackbar';
+import useSnackbar from 'components/Snackbar/useSnackbar';
 
-function GridContainerComponent({ storeData }) {
+const getDirective = (selectedFunction: IMenuItem, selectedColumnName: string | boolean) => {
+  return selectedFunction.getUsage({ selectedColumnName, selectedFunction });
+};
 
-  const [transformationPayload, setTransformationPayload] = useState<
-    Record<string, string | boolean>
-  >({
-    function: false,
-    column: false,
-  }); // whether i should apply transformation now or should pass data to add transformation step
+function GridContainerComponent({ storeData }) {
+  const [selectedFunction, setSelectedFunction] = useState({
+    option: {
+      getUsage({}) {
+        return '';
+      },
+      label: '',
+      supportedDataType: [''],
+      value: '',
+    },
+    supportedType: [''],
+  });
+  const [selectedColumn, setSelectedColumn] = useState(false);
 
-  const handleTransformationUpload = (valueToUpdate: string, newValue) => {
-    // valueToUpdate; // 'function', 'column'
+  const [snackbarState, setSnackbar] = useSnackbar();
 
-    setTransformationPayload((prev) => ({
-      function: valueToUpdate === 'function' ? newValue : prev.function,
-      column: valueToUpdate === 'column' ? newValue : prev.column,
-    }));
+  useEffect(() => {
+    if (snackbarState.open) {
+      setTimeout(() => {
+        setSnackbar(() => ({
+          open: false,
+        }));
+      }, 5000);
+    }
+  }, [snackbarState.open]);
 
-    // check if both function and column are selected
-    if (
-      (valueToUpdate == 'function' && transformationPayload.column) ||
-      (valueToUpdate == 'column' && transformationPayload.function)
-    ) {
-      // if yes, then do applyDirective
-      const directive = getDirective(newValue.option, transformationPayload.column);
-      // addDirectives()
+  /*
+   * In this useEffect we are handling apply transformation based on conditions
+   * whether to apply transformation directly without add transformation panel open
+   * or should pass data to add transformation step panel
+   */
+  useEffect(() => {
+    // checks if both function and column are selected
+    if (selectedColumn && selectedFunction) {
+      // If both function and column are selected, then apply transformation
```

Comment:
where do we handle transformations which need additional inputs other than function name and selected column?

Reply:
we will handle them in else block and these changes will be covered in other transformation PR's where we need additional inputs